x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -147,11 +147,27 @@ public PlanFactory visitEvalCommand(EsqlBaseParser.EvalCommandContext ctx) {
     public PlanFactory visitGrokCommand(EsqlBaseParser.GrokCommandContext ctx) {
         return p -> {
             String pattern = visitString(ctx.string()).fold().toString();
-            Grok result = new Grok(source(ctx), p, expression(ctx.primaryExpression()), Grok.pattern(source(ctx), pattern));
+            Grok.Parser grokParser = Grok.pattern(source(ctx), pattern);
+            validateGrokPattern(grokParser, pattern);
+            Grok result = new Grok(source(ctx), p, expression(ctx.primaryExpression()), grokParser);
             return result;
         };
     }
 
+    private void validateGrokPattern(Grok.Parser grokParser, String pattern) {
+        Map<String, DataType> definedAttributes = new HashMap<>();
+        for (Attribute field : grokParser.extractedFields()) {
+            String name = field.name();
+            DataType type = field.dataType();
+            DataType prev = definedAttributes.put(name, type);
+            if (prev != null) {
+                throw new ParsingException(
```

Comment:
Use an enhanced version of the constructor where the source is provided as well. As it is now, the error message looks like this `line -1:0: Invalid GROK pattern [%{NUMBER:foo} %{WORD:foo}]: the attribute [foo] is defined multiple times with different types` with an emphasis on `line -1:0` which is not really user friendly.